packages/opentelemetry-exporter-datadog/src/datadog.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as api from '@opentelemetry/api';
+import { ExportResult, NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { translateToDatadog } from './transform';
+import { AgentExporter, PrioritySampler, DatadogExporterConfig } from './types';
+import { URL } from 'url';
+
+/**
+ * Format and sends span information to Datadog Exporter.
+ */
+export class DatadogExporter implements SpanExporter {
+  private readonly _logger: api.Logger;
+  private readonly _exporter: typeof AgentExporter;
+  private _service_name: string;
+  private _env: string | undefined;
+  private _version: string | undefined;
+  private _tags: string | undefined;
+  private _url: string;
+  private _flushInterval: number;
+
+  constructor(config: DatadogExporterConfig = {}) {
+    this._url =
+      config.agent_url ||
+      process.env.DD_TRACE_AGENT_URL ||
```

Comment:
gotcha...I made a pass and tried to move as many things into consts and enums as possible. One or two places it causes some odd ts warnings that I tried to accomodate, but otherwise I think i've addressed the areas you've highlighted, let me know what you think